src/Symfony/Component/Validator/Constraints/RangeValidator.php

```diff
@@ -75,6 +75,18 @@ public function validate($value, Constraint $constraint)
                 ->setParameter('{{ limit }}', $this->formatValue($min, self::PRETTY_DATE))
                 ->setCode(Range::TOO_LOW_ERROR)
                 ->addViolation();
+
+            return;
+        }
+
+        if (null !== ($step = $constraint->step) && !\in_array($value, range($min, $max, $step), false)) {
```

Comment:
How will that work with dates?

Reply:
It does not work with date, that was not the feature, but we can do it like @Destroy666x said with a dateTimeRange.

Reply:
Well, I think we at least need to provide a solution that does not break when dates are passed.

Reply:
I guess we can do this and then provide the feature after if we need it ;).

Reply:
I think this should be reverted. Solution SHOULD break if DateTime is passed here and `step` is defined. Otherwise it silently works, which makes harder debugging.

It won't break existing code, since this is evaluated only if step is not null.